base/broadcast.jl

```diff
@@ -85,12 +113,13 @@ end
 # `newindex` above) for a particular array `A`, given the
 # broadcast_shape `shape`
 # `keep` is equivalent to map(==, indices(A), shape) (but see #17126)
-newindexer(shape, x::Number) = (), ()
-@inline newindexer(shape, A) = newindexer(shape, indices(A))
-@inline newindexer(shape, indsA::Tuple{}) = (), ()
-@inline function newindexer(shape, indsA::Tuple)
+newindexer(shape, x) = (), ()
+newindexer(shape, t::Tuple) = _newindexer(shape, (OneTo(length(t)),))
+@inline newindexer(shape, A::AbstractArray) = _newindexer(shape, indices(A))
```

Comment:
Equivalent, but there's little point calling it `broadcast_shape`, since `broadcast_shape(A) = indices(A)`. That function only comes into its own when passed multiple objects.

One concern with this generalization is that formerly, all tuples meant "tuples of array indices" and now they're sometimes acting like the array. I suspect we're now going to need some comments about what `Tuple` means in each particular signature.